### PR TITLE
Logger feature

### LINKED ISSUE #44: Getting ip address of the client
The `Connection` instance should be able to return the ip address of the client.

### PR DESCRIPTION
**Pull request by Nek-**
This adds the following:
- Simple logger that implements psr3
- Getting IP feature

This fixes #44 and #95

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -60,6 +60,11 @@ The `Websocket` instantiation take the following parameters:
 - `"127.0.0.1"`: the host you bind on, this is the default value and what you need most part of the time
 - `[]`: an (optional) array of configuration option documented in the [configuration reference](#configuration-reference)
 
+The `Connection` object have the following methods you can use:
```
Comment (folliked) [inline docs/index.md]: has ?

File: docs/index.md [hunk 2]
```diff
@@ -60,6 +60,11 @@ The `Websocket` instantiation take the following parameters:
...
+- `write($message, $opCode = Frame::OP_TEXT)`, you may change to `Frame::OP_BINARY` if you want to send binary data
+- `getIp()`, that returns the current IP
+- `getLogger()`, that returns the logger of woketo
+
 ### Use it your way
 
 Here are some helpers you can use depending on your use-case:
```
Comment (folliked) [inline docs/index.md]: There are ?

File: docs/index.md [hunk 3]
```diff
@@ -98,6 +103,20 @@ $defaultConfiguration = [
...
+
+```php
+<?php
+
+$server = new WebSocketServer(1337);
+$server->setLogger($myLogger);
+$server->run();
+```
+
+> Notice that this defined logger will be accessible from `Connection::getLogger()`.
```
Comment (folliked) [inline docs/index.md]: defines

File: src/Server/Connection.php [hunk 4]
```diff
@@ -101,6 +111,7 @@ private function processData($data)
             return;
         } catch (WebsocketException $e) {
             $this->messageProcessor->close($this->socketStream);
+            $this->logger->notice('Connection to ' . $this->getIp() . ' closed with error : ' . $e->getMessage());
```
Comment (folliked) [inline src/Server/Connection.php]: i prefer a sprintf in this case

File: src/Server/Connection.php [hunk 5]
```diff
@@ -135,11 +146,11 @@ protected function processMessage($data)
             } else {
                 // We wait for more data so we start a timeout.
                 $this->timeout = $this->loop->addTimer(Connection::DEFAULT_TIMEOUT, function () {
+                    $this->logger->notice('Connection to ' . $this->getIp() . ' timed out.');
```
Comment (folliked) [inline src/Server/Connection.php]: same

File: src/Server/WebSocketServer.php [hunk 6]
```diff
@@ -119,6 +125,8 @@ public function start()
         });
         $socket->listen($this->port);
 
+        $this->getLogger()->info('Listening on ' . $this->host . ':' . $this->port);
```
Comment (folliked) [inline src/Server/WebSocketServer.php]: same

File: src/Server/Connection.php (hunk 4 above)
Comment (Nek-) [inline src/Server/Connection.php]: just a preference, it's ok like that 👍 😉

File: src/Server/Connection.php (hunk 5 above)
Comment (Nek-) [inline src/Server/Connection.php]: same

File: src/Server/WebSocketServer.php (hunk 6 above)
Comment (Nek-) [inline src/Server/WebSocketServer.php]: same

**Pull request merged**